### PR TITLE
feat: TRAC-29-Add-support-for-Next.JS-to-the-Node-Distro

### PR DESCRIPTION
**Pull request by eugene-lumigo**
## Changes proposed

Adding support of Next.js
[Next.js OTEL](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry)
using 
[@vercel/otel](https://www.npmjs.com/package/@vercel/otel)

### CODE REVIEW

File: test/instrumentations/next/app/src/app/globals.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
+@tailwind base;
```
Comment (harelmo-lumigo) [inline test/instrumentations/next/app/src/app/globals.css:1]: In general, I'd like us to have as few files as possible related to test apps.
Let's remove this and `favicon.ico` too

File: test/instrumentations/next/app/src/app/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import type { Metadata } from "next";
```
Comment (harelmo-lumigo) [inline test/instrumentations/next/app/src/app/layout.tsx:1]: Unless the test app will break by removing this, let's remove this as well as the following `tsx` files.

File: test/instrumentations/next/next.test.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+async function getPortFree() {
+    return new Promise( res => {
+        const srv = net.createServer();
+        srv.listen(0, () => {
+            const port = srv.address().port
+            srv.close((err) => res(port))
+        });
+    })
+}
```
Comment (harelmo-lumigo) [inline test/instrumentations/next/next.test.ts:21]: IIRC this exists in another file - let's extract it to a shared util function please

**Review by harelmo-lumigo — APPROVED** [review]

File: test/instrumentations/next/app/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
+This is a [Next.js](https://nextjs.org) project bootstrapped with [`create-next-app`](https://nextjs.org/docs/app/api-reference/cli/create-next-app).
```
[the remark this answers is not in the document]
Comment (eugene-lumigo) [reply test/instrumentations/next/app/README.md:1]: Done

File: test/instrumentations/next/app/src/app/globals.css (hunk 1 above)
Comment (eugene-lumigo) [reply test/instrumentations/next/app/src/app/globals.css:1]: Done

File: test/instrumentations/next/app/src/app/layout.tsx (hunk 2 above)
Comment (eugene-lumigo) [reply test/instrumentations/next/app/src/app/layout.tsx:1]: Done

File: test/instrumentations/next/next.test.ts (hunk 3 above)
Comment (eugene-lumigo) [reply test/instrumentations/next/next.test.ts:21]: ok,

**Pull request merged**